narrative_llm_agent/agents/job.py

```diff
@@ -78,119 +75,69 @@ def __init_agent(self: "JobAgent") -> None:
-        def start_job(narrative_id: int, app_id: str, params: dict) -> str:
+        def start_job_tool(narrative_id: int, app_id: str, params: dict) -> str:
             """This starts a new job in KBase, running the given App with the given
             parameters in the given Narrative. If the app with app_id doesn't exist, this
             raises a AppNotFound error. If the narrative_id doesn't exist, or the user
             doesn't have write access to it, this raises a PermissionsError. If the
             parameters are malformed, or refer to data objects that do not exist, this
             returns a ValueError. If the app starts, this returns a JSON-formatted
             dictionary with cell_id and job_id fields."""
-            # print(info)
-            # inputs = json.loads(info)
-            # print(inputs)
-            # narrative_id = inputs["narrative_id"]
-            # params = inputs["app_params"]
-            # app_id = inputs["app_id"]
-            print("starting start_job tool")
-            print(f"narrative_id: {narrative_id}")
-            print(f"app_id: {app_id}")
-            print(f"params: {params}")
 
             if isinstance(params, str):
                 params = json.loads(params)
-            return self._start_job(
+            return start_job(
                 process_tool_input(narrative_id, "narrative_id"),
                 process_tool_input(app_id, "app_id"),
                 params,
+                ExecutionEngine(token=self._token),
+                NarrativeMethodStore(),
+                Workspace(token=self._token),
             )
 
         @tool("get-app-parameters", args_schema=AppInput, return_direct=False)
-        def get_app_params(app_id: str) -> str:
+        def get_app_params_tool(app_id: str) -> str:
             """
             This returns the set of parameters for a KBase app. This is returned as a
-            dictionary. If the app_id does not exist in KBase, this raises an AppNotFound
-            error.
+            stringified dictionary. If the app_id does not exist in KBase, this raises
+            an AppNotFound error.
             """
-            return self._get_app_params(app_id)
+            return json.dumps(get_app_params(app_id), NarrativeMethodStore())
 
         @tool("monitor-job", args_schema=JobInput, return_direct=False)
-        def monitor_job(job_id: str) -> str:
+        def monitor_job_tool(job_id: str) -> CompletedJob:
             """
             This monitors a running job in KBase. It will check the job status every 10 seconds.
             When complete, this returns the final job status as a JSON-formatted string. The
             final state can be either completed or error. This might take some time to run,
             as it depends on the job that is running.
             """
-            return self._monitor_job(process_tool_input(job_id, "job_id"))
+            ee = ExecutionEngine(token=self._token)
```

Comment:
I'm following the pattern here of having the invoking Agents (or other functions) handle their own clients. This gives a couple benefits.

1. auth gets centralized around the agents
2. tests get much easier with mocking and not having to do as much dependency injection.